### ISSUE TITLE
Increase of likelihood of crashes in past few days

### ISSUE THREAD

**Issue by alberttheprince** [citizenfx/fivem#2912]
### What happened?

The likelihood of users crashing while moving around the map (possibly because of streamed assets) has increased significantly over the past few days.

A report has been made on the forums: https://forum.cfx.re/t/sudden-increase-in-game-crashes-crashed-gta5-b3258-exe-sub-140777e18-type-errors/5280338

A number of client crash logs have been uploaded. We understand the vertex errrors, but these were all present a few days ago before any significant increase in crashes.

We have not made any changes to maps, clothing, or vehicles in the past few days, and see a significant increase in the number of client crashes occurring. Staying in one spot on the map will not cause users to crash, but people moving around will cause them to crash. If they try to connect to the server (usually after their character is loaded, so the user is teleported from the character select screen to the rest of the game world) they will crash immediately. 

Clearing the client cache will fix this.

### Expected result

No Crashes

### Reproduction steps

1. Stream Assets
2. Move around the map
3. Eventually crash
4. Can no longer join server without crashing
5. Clear client/FiveM cache, you are now able to connect to server without crashing
6. Eventually clients will crash 

### Importancy

Crash

### Area(s)

FiveM

### Specific version(s)

FiveM B3258

### Additional information

We use many streamed assets, and we understand there are some vertex issues. However, these crashes (usually lake-Edward-helium) have only recently started to occur. We have made no changes to maps and always attempt to optimize any added assets.

https://cfx.re/join/dppv4d

**Comment by LivedBackwardss** [comment]
Not sure if it's related to this or similar..

https://github.com/citizenfx/fivem/issues/2889
https://github.com/citizenfx/fivem/issues/2799

**Comment by alberttheprince** [comment]
> Not sure if it's related to this or similar..
> 
> #2889 #2799

If the issue is bad map makers, if there was a refund all these MLOs this wouldn't be an issue, but we're stuck with the cancer that is Tebex, meaning 0 accountability from these creators.

**Comment by LivedBackwardss** [comment]
> > Not sure if it's related to this or similar..
> > #2889 #2799
> 
> If the issue is bad map makers, if there was a way to refund all these MLOs this wouldn't be an issue, but we're stuck with the cancer that is Tebex, meaning 0 accountability from these creators.

Exactly my point, it never happened before and now out of the blue it happens. I've opened numerous tickets with creators and they say "the error does nothing" or don't reply nothing. This crash never happened before but some how in the update it starts to happen.

**Comment by alberttheprince** [comment]
> > > Not sure if it's related to this or similar..
> > > #2889 #2799
> > 
> > 
> > If the issue is bad map makers, if there was a way to refund all these MLOs this wouldn't be an issue, but we're stuck with the cancer that is Tebex, meaning 0 accountability from these creators.
> 
> Exactly my point, it never happened before and now out of the blue it happens. I've opened numerous tickets with creators and they say "the error does nothing" or don't reply nothing. This crash never happened before but some how in the update it starts to happen.

out of 10 creators I've contacted, 1 has fixed the issues, 1 said "I should just delete that map", and 1 has said "it shouldn't be a problem". 

I've reverted back to an older server artifact, but I'm not sure if that would be related. Hoping there's more insight from someone who would know.

**Comment by Legacy-TacticalGamingInteractive** [comment]
I can say with absolute certainty that those vertex errors (poly edge references) being fixed in real time adds up to the overhead that is required by a client while playing. THAT indeed causes crashes. Especially when you have multiple maps in one area causing this. Collision files that have broken poly edges will cause crashes more often than broken .ydr's for whatever the reason.

Most mappers do not know how to fix the problem. But even when you tell them how to do it such as using the Poly Edge Fixer tool by Smallo, they then take MONTHS if at all to get back to you. They don't give a shit because they have your money and they don't even run servers themselves.

I've dealt with this issue extensively.

There MAY be other things at play here, but I just wanted to mention my experience on the matter as well as being a definite cause of crashing in general for clients.

**Comment by LivedBackwardss** [comment]
> Most mappers do not know how to fix the problem. But even when you tell them how to do it such as using the Poly Edge Fixer tool by Smallo, they then take MONTHS if at all to get back to you. They don't give a shit because they have your money and they don't even run servers themselves.

Exactly why this is a huge issue. We pay for a product, they don't respond back and Tebex does nothing. This issue never happened in the past and now it happens everyday. I got over 100+ crashes from this error and still unable to find the cause. I removed most my stuff, at least the error could do is present the asset/resource thats causing it.

**Comment by alberttheprince** [comment]
> > > > > Not sure if it's related to this or similar..
> > > > > #2889 #2799
> > > > 
> > > > 
> > > > If the issue is bad map makers, if there was a way to refund all these MLOs this wouldn't be an issue, but we're stuck with the cancer that is Tebex, meaning 0 accountability from these creators.
> > > 
> > > 
> > > Exactly my point, it never happened before and now out of the blue it happens. I've opened numerous tickets with creators and they say "the error does nothing" or don't reply nothing. This crash never happened before but some how in the update it starts to happen.
> > 
> > 
> > out of 10 creators I've contacted, 1 has fixed the issues, 1 said "I should just delete that map", and 1 has said "it shouldn't be a problem".
> > I've reverted back to an older server artifact, but I'm not sure if that would be related. Hoping there's more insight from someone who would know.
> 
> Which artifact? It's not just us having the issue other people too.

10367 is stable. 

Chart from Txadmin shows the significant difference in crashes with (green box) 10367, and before (almost any build after). 

Timeouts are my fault. The initial crashes are users needing to clear their cache. Once cache is cleared, users no longer experience these issues forever on build 10367. 

![image](https://github.com/user-attachments/assets/55f22e1b-a41e-4114-a03d-a9ef372cfc79)


This issue is a server killer, so would avoid using anything until there is a fix.

**Comment by alberttheprince** [comment]
> I can say with absolute certainty that those vertex errors (poly edge references) being fixed in real time adds up to the overhead that is required by a client while playing. THAT indeed causes crashes. Especially when you have multiple maps in one area causing this. Collision files that have broken poly edges will cause crashes more often than broken .ydr's for whatever the reason.
> 
> Most mappers do not know how to fix the problem. But even when you tell them how to do it such as using the Poly Edge Fixer tool by Smallo, they then take MONTHS if at all to get back to you. They don't give a shit because they have your money and they don't even run servers themselves.
> 
> I've dealt with this issue extensively.
> 
> There MAY be other things at play here, but I just wanted to mention my experience on the matter as well as being a definite cause of crashing in general for clients.

This is likely irrelevant, as we're talking about 3-4 crashes a day vs. 2-300 crashes within a couple of hours based on server build.

**Comment by Legacy-TacticalGamingInteractive** [comment]
You are going to have to tell every single person who joins the server to clear the client cache? Geez

**Comment by alberttheprince** [comment]
> You are going to have to tell every single person who joins the server to clear the client cache? Geez

As far as I can tell, anyone who connects to a server with this issue will have to do so. I'm going to be getting more info on the nature of the crashes soon, but yeah basically if you connect to a server having the same issue as I am, the player needs to clear their cache.

**Comment by alberttheprince** [comment]
I've upgraded to the most recent artifact (windows:10822) and it seems to be working with no issues, although I'll need to test this with additional players. 

I tried to do this on previous builds, but was not able to reconnect. 

Leaving this open for now, as I honestly don't know what's changed, or if other's experience this.

**Comment by alberttheprince** [comment]
To add to what I said, it is still an issue in newer builds, but less so. I am trying to get more detailed reports, but it's hard to get players to get on and test due to the instability.

**Comment by LivedBackwardss** [comment]
> To add to what I said, it is still an issue in newer builds, but less so. I am trying to get more detailed reports, but it's hard to get players to get on and test due to the instability.

Still happens on the newer builds, shame really. Can't even play for x hours without crashing or the game crashing with no error. No player wants to play with this too.

**Comment by alberttheprince** [comment]
To speak to @Legacy-TacticalGamingInteractive 's point about the vertex errors, I've gone through and found all maps that could have these errors. So far I've found maps by 16 creators that had errors. 

Of these 16, only 2 creators acknowledged the issue and fixed it. of the 14 others, 4 said it wasn't a problem and would "eventually" fix it,  and the rest have no bothered responding. This includes large creators like k4mb1.

If this really is a problem with vertex errors, there's nothing we can do - and people with large MLO collections are out hundreds of dollars for broken resources that Tebex won't refund.

**Comment by FUst1ke** [comment]
> To speak to @Legacy-TacticalGamingInteractive 's point about the vertex errors, I've gone through and found all maps that could have these errors. So far I've found ~20 escrowed/paid maps by 16 creators that had errors.
> 
> Of these 16, only 2 creators acknowledged the issue and fixed it. Of the 14 others, 4 said it wasn't a problem and of those 4 two said they would "eventually" fix it, and the rest have not bothered responding. This includes large creators like k4mb1.
> 
> If this really is a problem with vertex errors, there's nothing we can do - and people with large MLO collections are out hundreds of dollars for broken resources that Tebex won't refund.

Hey, we have the same issue around here, would you make that creator list public, so we can be more careful when we chose maps from tebex?

**Comment by alberttheprince** [comment]
> > To speak to @Legacy-TacticalGamingInteractive 's point about the vertex errors, I've gone through and found all maps that could have these errors. So far I've found ~20 escrowed/paid maps by 16 creators that had errors.
> > Of these 16, only 2 creators acknowledged the issue and fixed it. Of the 14 others, 4 said it wasn't a problem and of those 4 two said they would "eventually" fix it, and the rest have not bothered responding. This includes large creators like k4mb1.
> > If this really is a problem with vertex errors, there's nothing we can do - and people with large MLO collections are out hundreds of dollars for broken resources that Tebex won't refund.
> 
> Hey, we have the same issue around here, would you make that creator list public, so we can be more careful when we chose maps from tebex?

Can you and @LivedBackwardss contact me on Discord? My Discord is noornahas. We can maybe compare resources/map files and see if there's any commonality between our servers having this issue.

Current creators with these issues:

Galaxis Studios (Player One Arcade), Royal LDS (Sherrif MLO), frizkstudios (Madrazo Mansion), prompt (only their older, free releases like Paleto Clinic), Flourine mapping (only their older, free releases), ROBBBS 3D (Freedom Tower MLO), Patoche (Plasma Kart, possibly their gang HQ), FM Shop (Binco and Fleeca MLOs), Pugs Mapping (Legion Square MLO), Juniors Interiors (Abra Kebabra, Belmont MLOs), Ai.Store (Sewers map/MLO) , Mineride Mapping (older, free interiors like Ponsonbys), K4MB1 (3 of their maps including the Casino, Paleto Bank, and their free Davis Library), UT MODZ (N Shop MLO), SoloTy FiveM (I saw the errors from their escrowed/free construction site MLO). 

I suspect there's more, but it takes time for me to find them.

**Comment by alberttheprince** [comment]
> I can say with absolute certainty that those vertex errors (poly edge references) being fixed in real time adds up to the overhead that is required by a client while playing. THAT indeed causes crashes. Especially when you have multiple maps in one area causing this. Collision files that have broken poly edges will cause crashes more often than broken .ydr's for whatever the reason.
> 
> Most mappers do not know how to fix the problem. But even when you tell them how to do it such as using the Poly Edge Fixer tool by Smallo, they then take MONTHS if at all to get back to you. They don't give a shit because they have your money and they don't even run servers themselves.
> 
> I've dealt with this issue extensively.
> 
> There MAY be other things at play here, but I just wanted to mention my experience on the matter as well as being a definite cause of crashing in general for clients.

We've gone through and used the tool to fix vertext issues, atm it doesn't seem to have made any effect on crashes. Players continue to crash regularly. 

There's about 12 maps we're waiting on the creators to fix themselves as they have escrowed ydr files, but it doesn't seem to have made a measurable effect on crashes.

**Comment by Legacy-TacticalGamingInteractive** [comment]
Its not surprising that its hard to tell. Because it mostly occurs when there are many other maps close to each other thus increasing the amount of overhead overall with the amount of models to be fixed in real time. ESPECIALLY when driving fast through an area. Thats why its a tough one to diagnose because it does not happen every single time.

Another big one is when they have collision files way behind the max limit of polys per object.

**Comment by DRAIKDev** [comment]
I have also big problemas with streaming assets crashes, maybe can something be done to at least have someting to debug about, maybe a list of assets were trying to load in the crash log or something similar that can help us to try identify the asset causing those crashes.

**Comment by alberttheprince** [comment]
> Its not surprising that its hard to tell. Because it mostly occurs when there are many other maps close to each other thus increasing the amount of overhead overall with the amount of models to be fixed in real time. ESPECIALLY when driving fast through an area. Thats why its a tough one to diagnose because it does not happen every single time.
> 
> Another big one is when they have collision files way beyond the max limit of polys per object.

Seems completely unrelated, players experiences crashes randomly and across the map. 

I appreciate the heads up on this issue. we've spent some time fixing vertex issues, but this seems completely unrelated. I don't think (at least from testing we've done) that this is related to loading collisions.

Additionally, for others who might be experiencing this, we don't seem to run into issues until the server has been up for 8 or so hours.

**Comment by LivedBackwardss** [comment]
No developer or response from the team with three issues opened and a lot of responses. Could you at least print the asset/model that's causing this issue, this is annoying dealing with it randomly.

**Comment by alberttheprince** [comment]
> No developer or response from the team with three issues opened and a lot of responses. Could you at least print the asset/model that's causing this issue to the console log or something before it crashes, this is annoying dealing with it randomly.

We've been given access to a development branch of the FiveM client to test issues potentially caused by how double archtypes are handled, which causes crashes consistently for our community. It seems to have fixed the issue for us, but others would need to confirm. 

To access it you need to add the line  **UpdateChannel=specific/feature/archetype_dupes** to your citizenfx.ini file in your FiveM files.

For anyone checking this, maybe try out this build with some players you know are experiencing crashing like we were.

**Comment by LivedBackwardss** [comment]
Still players receiving https://github.com/citizenfx/fivem/issues/2889 this error on release/beta. I'm not sure if there is a fix or what but it would be nice.

**Comment by alberttheprince** [comment]
> Still players receiving #2889 this error on release/beta. I'm not sure if there is a fix or what but it would be nice.

The fix is a only on canary ATM, so if if fixes the error you need fo have them switch

**Comment by alberttheprince** [comment]
My server averaging 10-20 players a day have all switched to canary, with the number of crashes dropping from 100-200 a day to nearly nothing.



@iridium-cfx there's no additional feedback I can give on this as your fixes seem to have completely solved all crashes we experienced. The error messages are very helpful (might be useful to keep them printing when pushed to release?) as well and I've been able to reach out to some creators who have fixed the problems. Although a lot of map makers also refuse to acknowledge/fix the issues.

Would be good to see this on release unless there's other issues people experienced with this build :)
